web/ui/src/Services/Service.js

```diff
@@ -141,9 +141,18 @@
         fetchInstances() {
             let deferred = $q.defer();
             resourcesFactory.v2.getServiceInstances(this.id)
-                .then(data => {
-                    // TODO - dont blow away existing instances
-                    this.instances = data.map(i => new Instance(i));
+                .then(results => {
+                    // console.log(`fetched ${data.length} instances for ${this.id}`);
+                    results.forEach(data => {
```

Comment:
Added the following
{noformat}
// new-ing instances will cause UI bouncing and force rebuilding
// of the popover. To minimize UI churn update/merge status info
// into exisiting instance ojects  
{noformat}